### ISSUE TITLE
Windows support

### ISSUE THREAD

**Issue by Redhawk18** [Smithay/calloop#161]
This is likely a giant undertaking, but this crate looks amazing. But I can't lock my ui library into not working on Windows. I use Linux daily but you still have to use libraries that work on every platform.

**Comment by notgull** [comment]
`polling` supports Windows. It's mostly the notification mechanism that needs to be ported.

**Comment by Redhawk18** [comment]
so Windows support is planned?

**Comment by elinorbgr** [comment]
I personally know nothing about Windows support. In particular wrt to ui library, I don't even know if it is actually possible to plug UI events from the Windows API into an event loop like `calloop`, I was under the impression that X1//Wayland were more the exception than the rule.

However, if someone who actually knows about this stuff is willing to contribute the necessary logic to make it worth, that would absolutely be welcome.

**Comment by Redhawk18** [comment]
> I personally know nothing about Windows support. In particular wrt to ui library, I don't even know if it is actually possible to plug UI events from the Windows API into an event loop like `calloop`, I was under the impression that X1//Wayland were more the exception than the rule.
> 
> However, if someone who actually knows about this stuff is willing to contribute the necessary logic to make it worth, that would absolutely be welcome.

doesn't winit have a event loop that works on all platforms?

**Comment by notgull** [comment]
A notification can be created by using [`PollerIocpExt::post`](https://docs.rs/polling/latest/x86_64-pc-windows-msvc/polling/os/iocp/trait.PollerIocpExt.html#tymethod.post). If someone has a motivating use case I can write this.

**Comment by Redhawk18** [comment]
> A notification can be created by using [`PollerIocpExt::post`](https://docs.rs/polling/latest/x86_64-pc-windows-msvc/polling/os/iocp/trait.PollerIocpExt.html#tymethod.post). If someone has a motivating use case I can write this.

It would make my ui work on windows then. I would feel stupid if my tui library had to use winit because of the event loop.

**Comment by elinorbgr** [comment]
> doesn't winit have a event loop that works on all platforms?

Not exactly, winit has a completely different event loop machinery on each platform it supports, tightly integrated in the specifics of that platform's GUI api. Something like that would be wildly out of scope for calloop.
